e2etest/GuestProxyAgentTest/TestCases/AddLinuxVMExtensionCase.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+using GuestProxyAgentTest.TestScenarios;
+using Azure.ResourceManager.Compute;
+
+namespace GuestProxyAgentTest.TestCases
+{
+    /// <summary>
+    /// Reboot VM test case
+    /// </summary>
+    public class AddLinuxVMExtensionCase : TestCaseBase
+    {
+        public AddLinuxVMExtensionCase() : base("AddLinuxVMExtensionCase")
+        { }
+        public AddLinuxVMExtensionCase(string testCaseName) : base(testCaseName)
+        { 
+        }
+
+        public override async Task StartAsync(TestCaseExecutionContext context)
+        {
+            var vmr = context.VirtualMachineResource;
+            var vmExtData = new VirtualMachineExtensionData(GuestProxyAgentTest.Settings.TestSetting.Instance.location)
+            {
+                Location = GuestProxyAgentTest.Settings.TestSetting.Instance.location,
+                Publisher = "Microsoft.CPlat.ProxyAgent",
+                ExtensionType = "ProxyAgentLinuxTest",
+                TypeHandlerVersion = "1.0",
+                AutoUpgradeMinorVersion = false,
+                EnableAutomaticUpgrade = false,
+                Settings = 
+                {
+                }
+            };
+            var result = await vmr.GetVirtualMachineExtensions().CreateOrUpdateAsync(Azure.WaitUntil.Completed, "ProxyAgentLinuxTest", vmExtData);
+            context.TestResultDetails = new GuestProxyAgentTest.Models.TestCaseResultDetails
+            {
+                CustomOut = result.Value.Data.ToString(),
+                StdOut = "Add Linux VM Extension case succeed.",
+                StdErr = "",
+                Succeed = true,
```

Comment:
what if the Add GPA vm extension with failed status?

Reply:
the test case and validation script would fail if gpa vm extension does not get installed correctly. the custom out would then show failed result.

Reply:
But you still mark this "success"?

Reply:
You need look into the details of "result" and fill the Succeed field, StdOut and StdErr if possible. You cannot hardcode Succeed to true no matter the result is success or failed.